src/SchemaRegistry.ts

```diff
@@ -165,39 +166,50 @@ export default class SchemaRegistry {
   }
 
   private async updateOptionsWithSchemaReferences(
-    options: SchemaRegistryAPIClientOptions | undefined,
     schema: ConfluentSchema,
+    options: SchemaRegistryAPIClientOptions | undefined,
   ) {
     const helper = helperTypeFromSchemaType(schema.type)
-    const referredSchemas = await this.getReferredSchemas(schema, helper)
-    return helper.updateOptionsFromSchemaReferences(options as ProtocolOptions, referredSchemas)
+    const referencedSchemas = await this.getreferencedSchemas(schema, helper)
+
+    const protocolOptions = this.asProtocolOptions(options)
+    return helper.updateOptionsFromSchemaReferences(referencedSchemas, protocolOptions)
+  }
+
+  private asProtocolOptions(options?: SchemaRegistryAPIClientOptions): ProtocolOptions | undefined {
+    if (!(options as LegacyOptions)?.forSchemaOptions) {
+      return options as ProtocolOptions | undefined
```

Comment:
Why do you need `| undefined`? Is that not implicit what happens if `options` is undefined?

Reply:
I am not sure - the output of asProtocolOptions should match the type of protocolOptions in helper.updateOptionsFromSchemaReferences